search-commons/src/main/java/no/unit/nva/search2/common/AggregationFormat.java

```diff
@@ -34,15 +36,37 @@ public static JsonNode apply(JsonNode aggregations, Map<String, String> definiti
         if (nonNull(aggregations)) {
             getAggregationFieldStreams(aggregations, definitions)
                 .map(AggregationFormat::getJsonNodeEntry)
-                .forEach(item -> objectNode.set(item.getKey(), fixNodes(item.getValue())));
+                .forEach(item -> objectNode.set(item.getKey(), fixNodes(item, item.getValue())));
         }
+        combineNotificationQueries(objectNode);
         return objectNode;
     }
 
-    private static JsonNode fixNodes(JsonNode node) {
+
+    private static void combineNotificationQueries(ObjectNode jsonNode) {
+        var notifications = JsonUtils.dtoObjectMapper.createObjectNode().arrayNode();
+        var keysToRemove = new ArrayList<String>();
+        for (Map.Entry<String, JsonNode> field : jsonNode.properties()) {
+            var fieldName = field.getKey();
+            if (isNotificationAggregation(field)) {
+                JsonNode value = field.getValue();
+                ((ObjectNode) value).put("key", fieldName);
+                notifications.add(value);
+                keysToRemove.add(fieldName);
+            }
+        }
+        keysToRemove.forEach(jsonNode::remove);
+        jsonNode.put("notifications", notifications);
+    }
+
+    private static boolean isNotificationAggregation(Entry<String, JsonNode> field) {
+        return field.getKey().toLowerCase(Locale.getDefault()).contains("notifications") && field.getValue().isObject();
+    }
```

Comment:
This is ugly, it should be possible to apply to Ticket aggregations only, but i feel this is the right place to do it. We can consider to create class TicketAggregation which would implement Aggregation and "apply" own rules to it.

Reply:
I tend to agree with  you, would you like to refactoring it?

Reply:
Yes, but it another pr.